app/src/main/java/mollie/tictactoe/GameHelper.java

```diff
@@ -0,0 +1,58 @@
+package mollie.tictactoe;
+
+import mollie.tictactoe.players.MobilePlayer;
+import ttt.game.Board;
+import ttt.game.GameEngine;
+import ttt.game.Marks;
+
+public class GameHelper {
+
+    private GameEngine mGameEngine;
+
+    public GameHelper(GameEngine game) {
+        mGameEngine = game;
+    }
+
+    public GameHelper() {
+    }
+
+    public GameEngine createGame() {
+        mGameEngine = new GameEngine(new MobilePlayer(Marks.X), new MobilePlayer(Marks.O), new Board(3));
+        return mGameEngine;
+    }
+
+    public String playMove(int moveLocation) {
+        mGameEngine = getGame();
+        mGameEngine.play(moveLocation);
+        return mGameEngine.board(moveLocation).toString();
+    }
+
+    public GameEngine getGame() {
+        if (isNotStarted()) {
+            createGame();
+        }
+        return mGameEngine;
+    }
+
+    public Board getBoard() {
+        if (isNotStarted()) {
+            return new Board(3);
+        }
+        return mGameEngine.showBoard();
+    }
+
+    private boolean isNotStarted() {
+        return mGameEngine == null;
+    }
+
+    public boolean gameIsOver() {
+        return mGameEngine.isOver();
+    }
+
+    public String getWinner() {
+        if (mGameEngine.isWon()) {
+            return mGameEngine.winningMark().toString();
+        }
+        return null;
```

Comment:
Ah! The infamous `null`! The crucible of the Java developer!
I will suggest a different signature: `public Optional<String> getWinner()`.
Instead of returning `null`, you return `Optional.empty()`.
The cool thing is that `Optional`, so you can use polymorphic dispatch on it! No need to "be careful" about those nulls. Give it a go and consider a *blog post* with the different parts of your app that were affected 😸